pkgs/development/python-modules/sh/default.nix

```diff
@@ -11,7 +11,7 @@ buildPythonPackage rec {
 
   # Disable tests that fail on Darwin
   # Some of the failures are due to Nix using GNU coreutils
-  patches = [ ./disable-broken-tests-darwin.patch ];
+  patches = [ ./disable-broken-tests.patch ];
```

Comment:
I think we could apply https://github.com/amoffat/sh/pull/468:
```diff
     sha256 = "1z2hx357xp3v4cv44xmqp7lli3frndqpyfmpbxf7n76h7s1zaaxm";
   };
 
-  # Disable tests that fail on Darwin
-  # Some of the failures are due to Nix using GNU coreutils
-  patches = [ ./disable-broken-tests-darwin.patch ];
+  patches = [
+    # Fix tests for Python 3.7. See: https://github.com/amoffat/sh/pull/468
+    (fetchpatch {
+      url = "https://github.com/amoffat/sh/commit/b6202f75706473f02084d819e0765056afa43664.patch";
+      sha256 = "1kzxyxcc88zhgn2kmfg9yrbs4n405b2jq7qykb453l52hy10vi94";
+      excludes = [ ".travis.yml" ];
+    })
+  ];

```

Reply:
I'll give it a try, it indeed seems to be exactly the same thing.

One question though: is a direct reference to a commit (part of a PR) on github something we can generally rely on existing in the future? Will such commits persist across a potential rewrite of the branch by the PR's author?

Reply:
The suggestion seems to work; to answer the other question myself: I just ran `find -type f | xargs grep "github.com.*patch"` which seems to indicate that the suggested pattern is quite wide-spread in nixpkgs; I'll change the PR as requested.